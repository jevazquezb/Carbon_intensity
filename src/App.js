import { Routes, Route } from 'react-router-dom';
import Header from './components/header';
import RegionList from './components/region_list';
import RegionDetails from './components/region_details';

function App() {
  return (
    <>
      <Header />
      <main>
        <Routes>
          <Route path="/region/:id" element={<RegionDetails />} />
          <Route path="/" element={<RegionList />} />
        </Routes>
      </main>
    </>
  );
}

export default App;
