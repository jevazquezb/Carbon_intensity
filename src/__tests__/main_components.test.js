import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/header';
import RegionList from '../components/region_list';
import RegionDetails from '../components/region_details';
import App from '../App';

describe('Render main components correctly', () => {
  test('Render Header component correctly', () => {
    const ContHeader = () => {
      <Router>
        <Header />
      </Router>;
    };
    const headerSnap = renderer.create(<ContHeader />).toJSON();
    expect(headerSnap).toMatchSnapshot();
  });

  test('Render RegionList component correctly', () => {
    const ContRegionList = () => {
      <Router>
        <RegionList />
      </Router>;
    };
    const regionListSnap = renderer.create(<ContRegionList />).toJSON();
    expect(regionListSnap).toMatchSnapshot();
  });

  test('Render RegionDetails component correctly', () => {
    const ContRegionDetails = () => {
      <Router>
        <RegionDetails />
      </Router>;
    };
    const regionDetailsSnap = renderer.create(<ContRegionDetails />).toJSON();
    expect(regionDetailsSnap).toMatchSnapshot();
  });

  test('Render App component correctly', () => {
    const ContApp = () => {
      <Router>
        <App />
      </Router>;
    };
    const appSnap = renderer.create(<ContApp />).toJSON();
    expect(appSnap).toMatchSnapshot();
  });
});
