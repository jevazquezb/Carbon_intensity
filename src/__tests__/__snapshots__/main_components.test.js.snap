// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render main components correctly Render App component correctly 1`] = `null`;

exports[`Render main components correctly Render Header component correctly 1`] = `null`;

exports[`Render main components correctly Render RegionDetails component correctly 1`] = `null`;

exports[`Render main components correctly Render RegionList component correctly 1`] = `null`;
